### PR TITLE
wasabi perps tvl integration

### PR DESCRIPTION
**Pull request by altugdurmaz**

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/wasabi: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:192:11

File: projects/wasabi/index.js [hunk 1]
```diff
@@ -1,21 +1,58 @@
...
+      onlyArgs: true,
+      fromBlock: 18810700,
+    });
+    resultVaults.push(...logs.map(log => log.vault));
+  }
+  return sumERC4626VaultsExport({ vaults: resultVaults, tokenAbi: 'asset', balanceAbi: 'totalAssets'});
+}
+
+const tvlBlastResult = await tvlBlast();
+const tvlEthResult = await tvlEth();
```
Comment (dev-wasabi) [inline projects/wasabi/index.js:49]: remove

File: projects/wasabi/index.js [hunk 2]
```diff
@@ -1,21 +1,58 @@
...
+
 module.exports = {
   ethereum: {
-    tvl
+    tvl: tvlEthResult
```
Comment (dev-wasabi) [inline projects/wasabi/index.js:53]: tvl: tvlEth

File: projects/wasabi/index.js [hunk 3]
```diff
@@ -1,21 +1,58 @@
...
+  },
+  blast: {
+    tvl: tvlBlastResult
```
Comment (dev-wasabi) [inline projects/wasabi/index.js:56]: tvl: tvlBlast

**Comment by llamatester** [comment]
Error while running adapter at projects/wasabi: 
        
 
  
    
    
    RangeError: data out-of-bounds (buffer=0x0000000000000000000000008e2b50413a53f50e2a059142a9be060294961e40, length=32, offset=64, code=BUFFER_OVERRUN, version=6.9.2)
    Truncated error stack:
        at makeError (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/ethers/lib.commonjs/utils/errors.js:126:21)
        at Array.forEach (<anonymous>)

**Comment by llamatester** [comment]
The adapter at projects/wasabi exports TVL: 
        
 
 
    blast                     18.65 M
    ethereum                  8.32 M
    
    total                    26.98 M

**Comment by g1nt0ki** [comment]
@altugdurmaz thanks for the PR, the code had a minor bug in the abi string, vault param was "indexed" but that was missing from the abi

**Comment by dev-wasabi** [comment]
Thank you for the fix! When will it be live?

**Pull request merged**